go-controller/pkg/ovn/gateway.go

```diff
@@ -265,7 +265,11 @@ func (gw *GatewayManager) GatewayInit(
 		// one node per zone, since ARPs for .1 will not go beyond local switch.
 		// This is being done to add the ICMP SNATs for .1 podSubnet that OVN GR generates
 		for _, subnet := range hostSubnets {
-			gwLRPIPs = append(gwLRPIPs, util.GetNodeGatewayIfAddr(subnet).IP)
+			if gw.netInfo.GatewayIP() != nil {
+				gwLRPIPs = append(gwLRPIPs, gw.netInfo.GatewayIP().IP)
+			} else {
+				gwLRPIPs = append(gwLRPIPs, util.GetNodeGatewayIfAddr(subnet).IP)
+			}
```

Comment:
nit: maybe we can store **always** the GW IP in the netInfo struct ? i.e. if one is present, you'd set the user-provided GW IP; if not, you'd set the `util.GetNodeGatewayIfAddr(subnet).IP` .

If would save us this if / else dance